src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingProvidersConfigImpl.java

```diff
@@ -0,0 +1,108 @@
+package io.stargate.sgv2.jsonapi.service.embedding.configuration;
+
+import io.quarkus.logging.Log;
+import io.smallrye.config.WithConverter;
+import io.stargate.embedding.gateway.EmbeddingGateway;
+import jakarta.annotation.Nullable;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public record EmbeddingProvidersConfigImpl(
+    Map<String, EmbeddingProviderConfig> providers, CustomConfig custom)
+    implements EmbeddingProvidersConfig {
+
+  public record EmbeddingProviderConfigImpl(
+      boolean enabled,
+      String url,
+      Map<AuthenticationType, AuthenticationConfig> supportedAuthentications,
+      List<ParameterConfig> parameters,
+      RequestProperties properties,
+      List<ModelConfig> models)
+      implements EmbeddingProviderConfig {
+
+    public record AuthenticationConfigImpl(boolean enabled, List<TokenConfig> tokens)
+        implements AuthenticationConfig {
+
+      public record TokenConfigImpl(String accepted, String forwarded) implements TokenConfig {}
+    }
+
+    public record ModelConfigImpl(
+        String name,
+        Optional<Integer> vectorDimension,
+        List<ParameterConfig> parameters,
+        Map<String, String> properties)
+        implements ModelConfig {
+
+      public ModelConfigImpl(
+          EmbeddingGateway.GetSupportedProvidersResponse.ProviderConfig.ModelConfig grpcModelConfig,
+          List<ParameterConfig> modelParameterList) {
+        this(
+            grpcModelConfig.getName(),
+            Optional.ofNullable(grpcModelConfig.getVectorDimension()),
+            modelParameterList,
+            grpcModelConfig.getPropertiesMap());
+      }
+    }
+
+    public record ParameterConfigImpl(
+        String name,
+        ParameterType type,
+        boolean required,
+        Optional<String> defaultValue,
+        Map<ValidationType, List<Integer>> validation,
+        Optional<String> help)
+        implements ParameterConfig {
+      public ParameterConfigImpl(
+          EmbeddingGateway.GetSupportedProvidersResponse.ProviderConfig.ParameterConfig
+              grpcModelParameter) {
+        this(
+            grpcModelParameter.getName(),
+            ParameterType.valueOf(grpcModelParameter.getType().name()),
+            grpcModelParameter.getRequired(),
+            Optional.ofNullable(grpcModelParameter.getDefaultValue()),
+            grpcModelParameter.getValidationMap().entrySet().stream()
+                .collect(
+                    Collectors.toMap(
+                            // @Hazel TODO ValidationType
```

Comment:
@Hazel-Datastax 
Left a TODO for you, like you said you may want to refactor the yaml again, and this is the place to modify.

Reply:
I think it's fine, I will not change the ValidationType, you can remove the TODO